### ISSUE TITLE
migrate to use apps/v1 for support kubernetes 1.16

### ISSUE THREAD

**Issue by DanielQujun** [goharbor/harbor-helm#403]
since kubernetes 1.16 deprecated DaemonSet, Deployment, StatefulSet, and ReplicaSet(in the extensions/v1beta1 and apps/v1beta2 API groups).
Deprecated APIs Removed In 1.16: https://kubernetes.io/blog/2019/07/18/api-deprecations-in-1-16/

we should migrate nginx deployment to use the apps/v1 API.

[templates/nginx/deployment.yaml](https://github.com/goharbor/harbor-helm/blob/master/templates/nginx/deployment.yaml)
```
{{- if ne .Values.expose.type "ingress" }}
apiVersion: extensions/v1beta1
kind: Deployment
```

**Comment by ywk253100** [comment]
@DanielQujun Thanks for explaining. A fix #377 has been merged into the master, closing this issue and your PR.

**Comment by deschmih** [comment]
@ywk253100 The merge is done, but it is still not available in release v1.2.2. Is is intended ?

**Comment by lhttjdr** [comment]
@deschmih It still doesn't work in release v1.2.3. 

I checked the git branch graph. It's merged into `master` branch, but the issue still exists in branch `1.2.0`.

I noticed there are several branches independent of `master`. It seems that they do not share such fixes in a quite long period. I do not know the strategy they manage this project. Perhaps it can not be solved in the near future.

**Comment by ywk253100** [comment]
Let's reopen it and cherry pick the fix into branch 1.2

**Comment by ywk253100** [comment]
The fix is merged into 1.2 branch and will be available in 1.2.4, closing this issue.
